src/tpm2_wrap.c

```diff
@@ -1014,25 +1019,38 @@ int wolfTPM2_SetAuthHandleName(WOLFTPM2_DEV* dev, int index,
         if (session->sessionHandle == TPM_RS_PW) {
             /* password based authentication */
             session->auth.size = handle->auth.size;
-            XMEMCPY(session->auth.buffer, handle->auth.buffer, handle->auth.size);
+            if (session->auth.size > sizeof(session->auth.buffer)) {
+                session->auth.size = sizeof(session->auth.buffer); /* truncate */
```

Comment:
In all these places where you truncate, should it actually be an error? Will the function return "success" but send back invalid data? If so, it would it be better to just return an error?

Reply:
The auth can be a password or hmac. Truncate seems like the right solution. If it is the wrong value then the device will return an error.